src/lib/mod.rs

```diff
@@ -77,59 +79,58 @@ impl HacksawResult {
     }
 }
 
-pub fn set_shape(conn: &xcb::Connection, window: xcb::Window, rects: &[xcb::Rectangle]) {
+pub fn set_shape<C: Connection>(conn: &C, window: xproto::Window, rects: &[xproto::Rectangle]) {
     shape::rectangles(
-        &conn,
-        shape::SO_SET as u8,
-        shape::SK_BOUNDING as u8,
-        0,
+        conn,
+        shape::SO::Set,
+        shape::SK::Bounding,
+        xproto::ClipOrdering::Unsorted,
         window,
         0,
         0,
         &rects,
-    );
+    ).unwrap().check().unwrap();
 }
 
-pub fn set_title(conn: &xcb::Connection, window: xcb::Window, title: &str) {
-    xcb::change_property(
-        &conn,
-        xcb::PROP_MODE_REPLACE as u8,
+pub fn set_title<C: Connection>(conn: &C, window: xproto::Window, title: &str) {
+    xproto::change_property(
+        conn,
+        xproto::PropMode::Replace,
         window,
-        xcb::ATOM_WM_NAME,
-        xcb::ATOM_STRING,
-        8,
+        xproto::AtomEnum::WM_NAME,
+        xproto::AtomEnum::STRING,
+        xproto::AtomEnum::STRING as u8,
```

Comment:
```suggestion
        8,
```

wait no this is wrong I forgot to git add the fix. Should be 8.